test/integration/taskBasedStatusUpdate.test.js

```diff
@@ -148,9 +177,38 @@ describe("Task Based Status Updates", function () {
         expect(res.body.userStatus.data.currentStatus).to.equal(userState.ACTIVE);
       });
 
-      it("Should change to ACTIVE state if the user is not ACTIVE.", async function () {
+      it("Should change to ACTIVE state if the user is not ACTIVE. ", async function () {
```

Comment:
`the user is not ACTIVE.` @heyrandhir if the user isn't active then do we change the status to ACTIVE?

Reply:
yes @vikhyat187 if the user has other task assigned on their name. then we will mark them active if their current status is IDLE.

Reply:
Got it.

Reply:
NIT: If we can create a visual diagram, then it will be easier for a new person to understand the flow much easier. 
@heyrandhir

Reply:
> NIT: If we can create a visual diagram, then it will be easier for a new person to understand the flow much easier. @heyrandhir

will do it @vikhyat187 . Thanks for suggesting